src/Raven.Client/Documents/Indexes/IAttachmentObject.cs

```diff
@@ -1,9 +1,20 @@
-﻿using System.IO;
+﻿using System;
+using System.IO;
 using System.Text;
+using Raven.Client.Documents.Attachments;
 
 namespace Raven.Client.Documents.Indexes
 {
-    public interface IAttachmentObject
+    public interface IAttachmentObject : IAttachmentObjectBase
```

Comment:
Shouldn't this be the other way around?
You introduced a new base interface, but that may be used.
Should introduce a new _derived_ one, no?